plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -1330,9 +1332,19 @@ void verifyChecksum(Plugin plugin, File pluginFile) {
         }
     }
 
+    @SuppressFBWarnings(value="WEAK_MESSAGE_DIGEST_SHA1", justification="Needed for legacy purposes.")
     private byte[] calculateChecksum(File pluginFile) {
         try (FileInputStream fin = new FileInputStream(pluginFile)) {
-            return DigestUtils.sha256(fin);
+            if("sha1".equals(getHashFunction())) {
+                return DigestUtils.sha1(fin);
+            }
+            else if("sha512".equals(getHashFunction())) {
+                return DigestUtils.sha512(fin);
+            } else if("sha256".equals(getHashFunction())) {
+                    return DigestUtils.sha256(fin);
+            } else {
+                return DigestUtils.sha256(fin);
```

Comment:
Should probably fail somewhere with a warning about unsupported hash function

Reply:
Probably best done in the `get` when it's a defined but unsupported value.

Reply:
It's a bit awkward because of the need to have lowercase toString()s for the enum, but I added the exception in here. The enum is protective of users, but I kept the exception in there as requested; it will at least protect against someone just trying add to the enum blindly, I suppose. See latest PR.